fieldtracing/fieldtracing.cpp

```diff
@@ -763,392 +760,191 @@ namespace FieldTracing {
+    * field line has reached a length of maxTracingDistance. In that case we call it DANGLING, it likely ended up in a loop
+    * somewhere (we don't call it "loop" to avoid confusion with the flux rop tracing). As long as we have not hit any of the above
+    * termination conditions, the type is called UNPROCESSED. We allow fieldTracingParameters.fullbox_max_incomplete_lines to remain
+    * UNPROCESSED when we exit the loop, that is a fraction of the total field lines we are tracing (number of cells x2 due to
+    * forward and backward tracing), as this allows substantial shortening of the total time spent due to the MPI_Allreduce calls
+    * occurring when we cross MPI domain boundaries. (NOTE: an algorithm doing direct task-to-task communication will likely be more
+    * efficient!) The connection type of the field line is a member of the enum TracingLineEndType and stored in cellFWConnection,
+    * cellBWConnection, cellConnection and the reduction arrays.
+    * enum TracingLineEndType {
+    *    UNPROCESSED,
+    *    CLOSED,
+    *    OPEN,
+    *    DANGLING,
+    *    N_TYPES
+    * };
+    * Once we have obtained a connection type for both directions, we parse the combinations and assign a value in the enum
+    * TracingPointConnectionType below to CellParams::CONNECTION that will be written out by the vg_connection DRO.
+    * enum TracingPointConnectionType {
+    *    CLOSED_CLOSED,
+    *    CLOSED_OPEN,
+    *    OPEN_CLOSED,
+    *    OPEN_OPEN,
+    *    CLOSED_DANGLING,
+    *    DANGLING_CLOSED,
+    *    OPEN_DANGLING,
+    *    DANGLING_OPEN,
+    *    DANGLING_DANGLING,
+    *    INVALID
+    * };
+    *
+    * Secondly, we are interested in finding out whether the seed point/DCCRG cell centre coordinate (again, too many fsgrid cells
+    * for sanity) are near/in a flux rope. As this relies also on tracing forward and backward along the field, we piggyback on the
+    * full box connection algorithm explained above.
+    * We trace along the field up to fieldTracingParameters.fluxrope_max_curvature_radii_to_trace*cellCurvatureRadius[n] and if
+    * within that tracing distance we have't extended further than
+    * fieldTracingParameters.fluxrope_max_curvature_radii_extent*cellCurvatureRadius[n] we are rolled up tightly enough to consider
+    * being close to a flux rope. We'll store the max extension reached into cellMaxExtension[n] for later fine-grained analysis.
+    * The arithmetic idea to avoid using yet more arrays:
+    * We use the cellFWConnection[n]/cellBWConnection[n]/cellConnection[n] (that can only be UNPROCESSED, CLOSED, OPEN or DANGLING,
+    * see TracingLineEndType enum above, ending with N_TYPES). As long as no full box tracing termination condition was reached
+    * we are at UNPROCESSED. If we exceed fieldTracingParameters.fluxrope_max_curvature_radii_to_trace*cellCurvatureRadius[n] OR
```

Comment:
could we also be dangling here?

Reply:
What it says on 919? Yes, but very unlikely, `maxTracingDistance` is bigger than `fieldTracingParameters.fluxrope_max_m_to_trace`. See also the check around line 1023.

Reply:
Should there be a check on parameter reading to ensure `maxTracingDistance` is at least as big as `fieldTracingParameters.fluxrope_max_m_to_trace`?